module4/src/com/epam/logic/TrainWorker.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.logic;
+
+import com.epam.entity.Train;
+
+
+public class TrainWorker {
+
+    private static int counter = 0;
+    private static Train yourTrain;
+
+    public static Train getTrainByNumber(Train[] trains, int inputNumber) {
+        for (int i = 0; i < trains.length; i++) {
+            if (TrainWorker.isTrainNumb(inputNumber, trains[i])) {
+                yourTrain = trains[i];
+                counter++;
+            }
+        }
+        if (counter == 0) {
+            yourTrain = null;
+        }
+        return yourTrain;
```

Comment:
It is better not to use counter for this purpose. Just assign **train to null by first line** and return train in the end